server/src/main/java/org/elasticsearch/common/xcontent/XContentBuilderExtension.java

```diff
@@ -61,4 +62,10 @@
      * @return a map of class name to transformer used to retrieve raw value
      */
     Map<Class<?>, XContentBuilder.HumanReadableTransformer> getXContentHumanReadableTransformers();
+
+    /**
+     * Used for plugging a transformer for a date or time type object into a String (or other
+     * encodable object)
+     */
```

Comment:
Can you please expand these javadocs to give examples? The generic Object to Object function doesn't really convey how someone would plugin in a date transform, or what it is used for (maybe linking to the timeValue methods where this matters would be good, at least with a @see for each method?).